modules/database/src/admin/schema.admin.ts

```diff
@@ -42,21 +47,38 @@ export class SchemaAdmin {
     const cndSchemas = await this.database
       .getSchemaModel('_DeclaredSchema')
       .model.findMany(
-        { 'modelOptions.conduit.cms': { $exists: false } },
+        {
+          $and: [
+            { 'modelOptions.conduit.cms': { $exists: false } },
+            { 'modelOptions.conduit.permissions.extendable': true },
+          ],
+        },
         {
           select: 'name extensions',
           sort: {
             updatedAt: 1,
           },
         },
       );
-    return { schemas: cmsSchemas.concat(cndSchemas) };
+    return {
+      schemas: cmsSchemas
+        .concat(cndSchemas)
+        .map((schema: ConduitDatabaseSchema) => ({
+          ...schema,
+          extensions: schema.extensions.filter(ext => ext.ownerModule === 'database'),
+        })),
+    };
   }
 
   async importSchemas(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
     const schemas = call.request.params.schemas;
     for (const schema of schemas) {
-      const isCmsSchema = schema.modelOptions;
+      const found = await this.database
+        .getSchemaModel('_DeclaredSchema')
+        .model.findOne({ name: schema.name });
+      const isCmsSchema = found
+        ? !!found.modelOptions.conduit.cms
+        : !!schema?.modelOptions.conduit.cms;
```

Comment:
Right-hand case should be `!!schema.modelOptions?.conduit?.cms` (or similar).
Schema is always gonna be defined as you're looping through the schema array, but any nested object fields could (and technically should) be missing for module schemas.